src/GovUk.Education.ExploreEducationStatistics.Data.Api/Controllers/TableBuilderController.cs

```diff
@@ -121,6 +121,25 @@ public async Task<ActionResult<TableBuilderResultViewModel>> QueryForTableBuilde
             return actionResult;
         }
 
+        [HttpGet("tablebuilder/release/data-block/{dataBlockParentId:guid}")]
+        public async Task<ActionResult<Dictionary<string, List<LocationAttributeViewModel>>>> QueryForTableBuilderWithGeoJsonResult(
```

Comment:
Out of interest, do we not need an equivalent method in Admin's TableBuilderController?  Also, I know that we cache DataBlock results in Admin (see Admin's TableBuilderController):

```
        [BlobCache(typeof(DataBlockTableResultCacheKey))]
        private async Task<Either<ActionResult, TableBuilderResultViewModel>> GetReleaseDataBlockResults(
            DataBlockVersion dataBlockVersion,
            long? boundaryLevelId,
            CancellationToken cancellationToken)
        {
            return await _userService
                .CheckCanViewReleaseVersion(dataBlockVersion.ReleaseVersion)
                .OnSuccess(_ => _tableBuilderService.Query(releaseVersionId: dataBlockVersion.ReleaseVersionId,
                    dataBlockVersion.Query,
                    boundaryLevelId,
                    cancellationToken));
        }
```

I wonder if we need to look at caching in Admin too?

Reply:
I queried this with @amyb-hiveit, who implied that we could possibly use the same endpoint for both.

Reply:
I think you will still need another endpoint creating in the backend, because (as I think you're already aware) while the frontend sometimes uses `dataApi` to make Admin API calls, the backend explicitly has separate endpoints.

So `QueryForTableBuilderResults` in the Data API's `TableBuilderController` has a twin `QueryForDataBlock` in Admin's `TableBuilderController`. The only difference between the two, as far as I'm aware, is the Admin's prefixes `data/` to the endpoint path and has a permissions check.

The frontend utilises both of these endpoints via a sole method, `tableBuilderService.ts`'s `getDataBlockTableData` - basically `dataApi` (and confusingly) when called from admin, actually calls admin. This is to allow the component `DataBlockTabs` (and possibly others?) to be used in `src/ees-admin` and `src/ees-frontend`.